workflows/metaval.nf

```diff
@@ -85,15 +92,54 @@ workflow METAVAL {
         ch_extract_reads.diamond_tsv,
         )
         ch_versions            = ch_versions.mix( TAXID_READS.out.versions )
-    }
-
-    /*
-        SUBWORKFLOW: Screen pathogens
-    */
-    ch_reference = Channel.fromPath( params.pathogens_genome, checkIfExists: true)
+        // Remove empty fastq files produced by extracting reads for user defined taxIDs
+        if (params.extract_kraken2_reads && params.taxid) {
+            RM_EMPTY_KRAKEN2(file("${params.outdir}/extracted_reads/kraken2"))
+        }
+        if (params.extract_centrifuge_reads && params.taxid) {
+            RM_EMPTY_CENTRIFUGE(file("${params.outdir}/extracted_reads/centrifuge"))
+        }
+        if (params.extract_diamond_reads && params.taxid) {
+            RM_EMPTY_DIAMOND(file("${params.outdir}/extracted_reads/diamond"))
+        }
+
+        // SUBWORKFLOW: DE NOVO
+
+        // Filter out empty FASTQ files
+        ch_taxid_reads = TAXID_READS.out.reads.filter { meta, reads ->
+            def files = reads instanceof List ? reads: [ reads ] // type checking to ensure that reads is a list
+            files.findAll { filepath ->
+                def file = new java.io.File ( filepath.toString() ) //create file objects
+                file.size() > 0
+            }
+        }
+        // Prepare reads for de-novo assembly
+        ch_denovo_input = ch_taxid_reads.branch { meta, reads ->
+            def files = reads instanceof List ? reads : [reads] // Ensure reads is a list
+            def count = files[0].countFastq()
+            shortreads_spades: meta.instrument_platform != 'OXFORD_NANOPORE' && count > params.min_read_counts
+                return [ meta, reads, [], [] ]
+            longreads_denovo: meta.instrument_platform == 'OXFORD_NANOPORE' && count > params.min_read_counts
+                return [ meta, reads ]
+        }
+
+        // short reads de novo assembly
+        if (! params.skip_shortread_denovo ) {
+            SPADES(ch_denovo_input.shortreads_spades, [], [])
+            ch_versions             = ch_versions.mix( SPADES.out.versions.first() )
+        }
+        // long reads de novo assembly
+        if (! params.skip_longread_denovo ) {
+            FLYE( ch_denovo_input.longreads_denovo, "--nano-corr" )
```

Comment:
Should `--nano-corr` be in modules.config instead?

Reply:
Very good suggestion.

Reply:
I see! 
But then you can set it in `nextflow.config` as `flye_mode='--nano-corr'` and pass this as `params.flye_mode` in `metaval.nf`

Reply:
Done!